install.sh

```diff
@@ -164,6 +166,35 @@ if [ "$SENTRY_DATA_NEEDS_MIGRATION" ]; then
     "mkdir -p /tmp/files; mv /data/* /tmp/files/; mv /tmp/files /data/files; chown -R sentry:sentry /data"
 fi
 
+
+if [ ! -f "$RELAY_CREDENTIALS_JSON" ]; then
+    echo ""
+    echo "Generating Relay credentials..."
+
+    $dcr --user $(id -u) relay --config /etc/relay credentials generate --overwrite
+    chmod a+r $RELAY_CREDENTIALS_JSON
+    # display the contents of the relay directory (for debug purposes)
+    ls -al ./relay
+    CREDENTIALS=$(sed -n 's/^.*"public_key":[[:space:]]"\([a-zA-Z0-9_-]*\)".*$/\1/p' "$RELAY_CREDENTIALS_JSON")
```

Comment:
I think the space here should be followed by a `*` as it is optional in JSON (or can be repeated). Also note the `*` -> `+` change for the actual credentials value matcher.
```suggestion
    CREDENTIALS=$(sed -n 's/^.*"public_key":[[:space:]]*"\([a-zA-Z0-9_-]+\)".*$/\1/p' "$RELAY_CREDENTIALS_JSON")
```

Reply:
Did you test it on Mac? 
Maybe we have a different version of sed installed because it doesn't work on my Mac ( I had quite a few problems with the Mac sed) it seems to only recognise a very limited regex set `+` doesn't seem to work. 

I could have used `[a-zA-Z0-9_-][a-zA-Z0-9_-]*` but it didn't seem to be worth my while. 

I added the optional number of spaces suggested `[[:space:]]*` but since this is the output of a program generated file I don't think we really need to support all possible correct json files ( which we don't anyway, since we don't support spaces before `:` or the key and the value being spread on two different lines.

Reply:
>Did you test it on Mac?

Nope as I'm using Windows + Linux. Seems like `+` requires the extended regexp set which needs different flags for Linux and MacOS (`-r` and `-E` respectively). So instead we can do `[a-zA-Z0-9_-]\{1,\}`.

I think this is important because it expresses our intent more clearly. I'd prefer a fixed length if the credentials string has a fixed length like a hash for precision.

Reply:
>I added the optional number of spaces suggested [[:space:]]* but since this is the output of a program generated file I don't think we really need to support all possible correct json files ( which we don't anyway, since we don't support spaces before : or the key and the value being spread on two different lines).

Good point, we should probably add a `[[:space:]]*` before the colon too. The output of the program is JSON and the formatting of the JSON file can change with a newer Rust version, a different JSON library or some other config change in that program so I think trying to cover these basics are important.